### PR TITLE
Remove package list after a build

### PR DESCRIPTION
**Pull request by azman0101**
For security reason, is it not safer to remove some packages after build (compilers for example) ?

### CODE REVIEW

**Comment by drybjed** [comment]
First, to answer your question, `debops.backporter` was not designed to remove anything. This role was created when the GitLab project started to require Ruby 2.1 packages. To enable installation of GitLab on Debian Wheezy, which at the time was the Stable distribution, I written this "skeletal role" with an idea in mind that you could define it as a role dependency and on special temporary VM, after checking that some package version was not available in APT, it would get the newer sources from Debian Testing and build backported `.deb` packages. You could then put them in a local APT repository and offer them on your regular production hosts, on which the `debops.backporter` would detect the newer packages and not repeat the whole process again. So in essence, you would create a temporary build host and remove it after the fact - nothing to remove.

Buuut, the idea was flawed from the start. Pretty quick Jessie became Stable, and the need to backport newer packages to Wheezy became moot almost overnight. Now this role is in a few role dependencies like a dead weight dragged around from the "old days". Especially that the core design of role dependencies in DebOps has evolved from "hard dependencies" in `meta/main.yml` which need to be installed and run always, to "soft dependencies" as roles that run in the playbook before your main role and to which you just pass variables with required configuration.

In other words, this role is obsolete. The concept of an easy to use backporter is sound, but the implementation is deeply flawed and basically needs to be shed entirely from the project. If you are looking for an alternative, you should check out [Deb-o-Matic](https://debomatic.github.io/) or [rebuildd](https://julien.danjou.info/projects/rebuildd) which implement the idea of an automated package backporting in a more sane way. I plan to create a role for `rebuildd` + `pbuilder` packages to allow creation of a `.deb` build infrastructure, but I haven't had time to get around that yet - other roles took prrecedence for now.

**Comment by azman0101** [comment]
Ok thank for your precise answer.

It would be great to add this point about backporter obsolescence in the
doc, isn't it ?

Thanks for your advice.
Le 14 juin 2016 11:49 PM, "Maciej Delmanowski" <notifications@github.com> a
écrit :

**Comment by drybjed** [comment]
You're right, README should point that out. I'll update it after I'm back at work in a few days, and I'll look into finally removing the role dependence from other roles.

**Comment by drybjed** [comment]
This role has been deprecated and will not be included in the unified `git` repository, therefore I'm closing this pull request.
